src/app/item-page/media-viewer/media-viewer.component.ts

```diff
@@ -95,37 +101,44 @@ export class MediaViewerComponent implements OnDestroy, OnInit {
       ...(this.mediaOptions.video ? ['audio', 'video'] : []),
     ];
     this.thumbnailsRD$ = this.loadRemoteData('THUMBNAIL');
-    this.subs.push(this.loadRemoteData('ORIGINAL').subscribe((bitstreamsRD: RemoteData<PaginatedList<Bitstream>>) => {
-      if (bitstreamsRD.payload.page.length === 0) {
-        this.isLoading = false;
-        this.mediaList$.next([]);
-      } else {
-        this.subs.push(this.thumbnailsRD$.subscribe((thumbnailsRD: RemoteData<PaginatedList<Bitstream>>) => {
-          for (
-            let index = 0;
-            index < bitstreamsRD.payload.page.length;
-            index++
-          ) {
-            this.subs.push(bitstreamsRD.payload.page[index].format
-              .pipe(getFirstSucceededRemoteDataPayload())
-              .subscribe((format: BitstreamFormat) => {
-                const mediaItem = this.createMediaViewerItem(
-                  bitstreamsRD.payload.page[index],
-                  format,
-                  thumbnailsRD.payload && thumbnailsRD.payload.page[index],
-                );
-                if (types.includes(mediaItem.format)) {
-                  this.mediaList$.next([...this.mediaList$.getValue(), mediaItem]);
-                } else if (format.mimetype === 'text/vtt') {
-                  this.captions$.next([...this.captions$.getValue(), bitstreamsRD.payload.page[index]]);
-                }
-              }));
-          }
+    this.subs.push(this.loadRemoteData('ORIGINAL')
+      .subscribe((bitstreamsRD: RemoteData<PaginatedList<Bitstream>>) => {
+        if (bitstreamsRD.payload.page.length === 0) {
           this.isLoading = false;
-          this.changeDetectorRef.detectChanges();
-        }));
-      }
-    }));
+          this.mediaList$.next([]);
+        } else {
+          this.subs.push(this.thumbnailsRD$.subscribe((thumbnailsRD: RemoteData<PaginatedList<Bitstream>>) => {
+            for (
+              let index = 0;
+              index < bitstreamsRD.payload.page.length;
+              index++
+            ) {
+              this.subs.push(this.isAuthorized(bitstreamsRD.payload.page[index]).subscribe(
+                isAuthorize => {
+                  if (isAuthorize) {
+                    this.subs.push(bitstreamsRD.payload.page[index].format
+                      .pipe(getFirstSucceededRemoteDataPayload())
+                      .subscribe((format: BitstreamFormat) => {
```

Comment:
@VictorDuranEscire : This looks like you have several nested "subscribe()" calls here.  This subscribe is nested under the "subscribe()" on line 116, and that one is nested under the "subscribe()" on line 110.

This is generally a bad practice in Angular / RxJS.  See for example:
* https://blog.brecht.io/rxjs-best-practices-in-angular/#avoiding-nested-subscribes 
* https://stackoverflow.com/questions/55416011/how-can-i-avoid-multiple-nested-subscriptions-using-rxjs-operators
* https://medium.com/ngconf/why-you-shouldnt-nest-subscribes-eafbc3b00af2 
Those provide some basic examples on how to clean up this code a bit.